### PR TITLE
Revise the usage of the terms `command` and `task` for consistency

### PR DESCRIPTION
**Pull request by tigitz**
Fix issue https://github.com/jolicode/castor/issues/231

I went with the following approach:
- Castor is a **task** runner, providing built-in **helper** functions for common task logics. (Reasoning: `run()` is a helper the same way `with()` is, provided functions are not necessarily **commands**)
-  **Commands** are what the `run()` helper is running, processes are the result of running a command.
- ~~Avoid using **function** broader term when really it's about castor built-in **helpers**~~

I'm still unsure how to rename the internal `TaskCommand`, `RepackCommand` etc. Because they are in essence Symfony command but are displayed alongside other tasks when doing `bin/castor`.

**Todo**:
- [x] Settle on internal naming for "Command" class when using Symfony Command class ( `TaskCommand`, `RepackCommand`, etc.)
- [x] List the BC Breaks (there's none)

### CODE REVIEW

File: src/Console/Command/Init.php [hunk 1]
```diff
@@ -14,7 +14,7 @@
     name: 'init',
     description: 'Initializes a new Castor project',
 )]
-class InitCommand extends Command
+class Init extends Command
```
Comment (TheoD02) [inline src/Console/Command/Init.php:17]: Why `Init` rather than `InitTask` ? This allows you to say that it is an initiation task. `Init` may seem too generic WDYT ?

File: src/Console/Command/Repack.php [hunk 2]
```diff
@@ -14,7 +14,7 @@
 /**
  * @internal
  */
-class RepackCommand extends Command
+class Repack extends Command
```
Comment (TheoD02) [inline src/Console/Command/Repack.php:17]: Same here keep `RepackTask` ?

File: src/Console/Command/Init.php (hunk 1 above)
Comment (tigitz) [reply src/Console/Command/Init.php:17]: Yeah I thought about it but I was wondering how to rename the `TaskCommand`  then and didn't find a consistent naming for the 3 of them that wouldn't be confusing.

Maybe `RepackTask`, `InitTask` and `BaseTask`?

Comment (TheoD02) [reply src/Console/Command/Init.php:17]: Yes, I think it would be more coherent to name them that way 😄

Comment (pyrech) [reply src/Console/Command/Init.php:17]: IMO these are internal code so it should keep respecting Symfony standards and be named command.

Comment (tigitz) [reply src/Console/Command/Init.php:17]: @pyrech I thought about it too but I'm afraid this implicit understanding will still generate ambiguity and confusion, even for experienced contributors. Case in point: https://github.com/jolicode/castor/pull/218#discussion_r1446018497

Since castor is deeply coupled with symfony console and command, we could do `TaskSymfonyCommand`, `InitSymfonyCommand`, `RepackSymfonyCommand` ? Or `TaskConsoleCommand` etc. ? 

To really reflect which "command" it's about ?

**Comment by tigitz** [comment]
PR is in a reviewable state, I've updated the PR description with a remaining list of TODOs.

**Comment by pyrech** [comment]
I fully agree we need to revise the usage of the terms command and task in both userland code and documentation so I'm :+1: for this.

But I still don't agree on the renaming of the internal parts (mainly related to Symfony Console component). It does not impact users and should respect Symfony standards to ease contributions and maintenance IMO.

About the `function` / `helper` subject, I'm also not really convinced by the renaming. As you stated in the README, Castor "leverages PHP's scripting capabilities" and we are used to the term `function` in PHP. Thus calling built-in functions "function" made sense to me. But I don't have strong argument on this subject so let's see what others thinks on this one. :slightly_smiling_face:

**Comment by lyrixx** [comment]
> But I don't have strong argument on this subject so let's see what others thinks on this one.

I would also add that usually people tend to use `functions.php` in their code base. at least in syfmfony. I prefer to stick with this standard

**Comment by tigitz** [comment]
@pyrech @lyrixx Understood, I'll revert the function -> helpers renaming and leave the internal "Command" classes as is.

**Comment by lyrixx** [comment]
Thanks, and could you also rebase?

**Comment by tigitz** [comment]
@lyrixx @pyrech updates done, PR ready :+1: , cs failure unrelated

File: README.md [hunk 3]
```diff
@@ -116,8 +120,8 @@ Discover more by reading the docs:
 
 * [Installation and Autocomplete](doc/01-installation.md)
 * [Basic Usage](doc/02-basic-usage.md)
-* [Executing commands](doc/03-run.md)
-* [Command Arguments](doc/04-arguments.md)
+* [Executing Commands with `run()`](doc/03-run.md)
```
Comment (lyrixx) [inline README.md:123]: Juste a question: maybe we can say "process" here? (Don't change now :) )

File: doc/03-run.md [hunk 4]
```diff
@@ -18,15 +17,15 @@ function foo(): void
...
+You can pass a string or an array of string for this function. When passing a
 string, arguments will not be escaped - use it carefully.
 
 ## Process object
 
 Under the hood, Castor uses the
 [`Symfony\Component\Process\Process`](https://github.com/symfony/symfony/blob/6.3/src/Symfony/Component/Process/Process.php)
-object to run the command. The `run()` function will return this object. So
-you can use the API of this class to interact with the process:
+object to execute the command. The `run()` function will return this object. So
```
Comment (lyrixx) [inline doc/03-run.md:27]: yes, IHMO "process" is better in this chapter.

File: src/Console/Application.php [hunk 5]
```diff
@@ -121,7 +121,7 @@ public function getSymfonyStyle(): SymfonyStyle
         return $this->symfonyStyle ?? throw new \LogicException('SymfonyStyle not available yet.');
     }
 
-    public function getCommand(): Command
+    public function getTask(): Command
     {
         return $this->command ?? throw new \LogicException('Command not available yet.');
     }
```
Comment (lyrixx) [inline src/Console/Application.php:127]: I'm uncomfortable with this change.
If we go this way, the exception message must be updated too!

But according to the signature, I prefer to keep `getCommand`. And so, I prefer to revert changes in src/Attribute/AsOption.php and src/Attribute/AsTaskArgument.php

File: src/Console/Application.php [hunk 6]
```diff
@@ -175,7 +175,7 @@ private function initializeApplication(InputInterface $input): array
         if (class_exists(\RepackedApplication::class)) {
             $functionsRootDir = \RepackedApplication::ROOT_DIR;
         }
-        // Find all potential commands / context
+        // Find all potential tasks / context
```
Comment (lyrixx) [inline src/Console/Application.php:178]: let's remove this comments, it does not bring anything useful

File: src/Console/Command/RepackCommand.php [hunk 7]
```diff
@@ -30,7 +30,7 @@ protected function configure(): void
     protected function execute(InputInterface $input, OutputInterface $output): int
     {
         if (str_starts_with(__FILE__, 'phar:')) {
-            throw new \RuntimeException('This command cannot be run from a phar. You must install castor with its sources.');
+            throw new \RuntimeException('This task cannot be run from a phar. You must install castor with its sources.');
```
Comment (lyrixx) [inline src/Console/Command/RepackCommand.php:33]: In this case, it's not a task, but a command. Could you revert it?

File: src/Console/Command/TaskCommand.php [hunk 8]
```diff
@@ -36,12 +36,12 @@ public function __construct(
         $this->setDescription($taskAttribute->description);
         $this->setAliases($taskAttribute->aliases);
 
-        $commandName = $taskAttribute->name;
+        $taskName = $taskAttribute->name;
```
Comment (lyrixx) [inline src/Console/Command/TaskCommand.php:39]: Note for myself (and the next review): I had a long conversation with the other me, and I think this change is OK even if I'm not convinced.

File: src/GlobalHelper.php [hunk 9]
```diff
@@ -77,9 +77,9 @@ public static function getSymfonyStyle(): SymfonyStyle
         return self::getApplication()->getSymfonyStyle();
     }
 
-    public static function getCommand(): Command
+    public static function getTask(): Command
```
Comment (lyrixx) [inline src/GlobalHelper.php:80]: same here, I think we should revert this change

**Review by lyrixx — COMMENTED** [review]
Thanks for the hard work. I let some comments, but I think it's almost ok.

**Review by pyrech — COMMENTED** [review]
Just left some more comments. Thanks for all the modifications :pray:

**Comment by tigitz** [comment]
Updated and ready to be reviewed.

@lyrixx I wasn't sure if you expected the renaming from command to process as you said "don't do it" but then "yes "process" is better in this chapter" so I went and did it. Feel free to clarify, I'll quickly revert it if needed.

**Review by pyrech — APPROVED** [review]
Thanks for your work on this topic :pray:
